### ISSUE TITLE
Unable to deploy model with loadbalancer enabled.

### ISSUE THREAD

**Issue by msinha251** [microsoft/DeepSpeed-MII#180]
Hi Team, I am trying to run deploy the model over fast api. But getting  below error. If i disable the restful_api by commenting below lines
```
"enable_restful_api": True,
"restful_api_port": input_data.restful_api_port,
"hostfile": "./hostfile",
```
then i am able to deploy model successfully but unable to query the model getting `this event loop is already running` (already tried all discussed approaches but nothing worked), therefore now i am trying to get it deploy with restful api:

error:
`ValueError: No slot '1' specified on host 'localhost'`

My code is as follows:
```
from fastapi import FastAPI, HTTPException
from pydantic import BaseModel
import mii
import logging

logging.basicConfig(
    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
)

app = FastAPI()

class QueryInput(BaseModel):
    deployment_name: str
    query: list[str]

class DeployInput(BaseModel):
    model: str
    deployment_name: str
    tensor_parallel: int = 8
    dtype: str = "torch.float16"
    enable_restful_api: bool = True
    restful_api_port: int = 8004

    class Config:
        extra = "forbid"

# deployed_models = {}
@app.post("/generate/")
async def generate_text(input_data: QueryInput):
    logging.info(f"Received request for deployment '{input_data.deployment_name}'")
    deployment_name = input_data.deployment_name
    try:
        generator = mii.mii_query_handle(deployment_name)
        logging.info(f"Generating text for deployment '{deployment_name}'")
        logging.info(f"Type of generator: {type(generator)}")
        logging.info(f"Input: {input_data.query}")   
        generator = mii.mii_query_handle(deployment_name)
        response = generator.query({'query': input_data.query})
        logging.info(f"Text generated for deployment '{deployment_name}'")
        logging.info(f"Response: {response}")
        return response
    except Exception as e:
        logging.error(f"Error generating text for deployment '{deployment_name}'")
        logging.error(f"Error: {e}")
        raise HTTPException(status_code=500, detail=str(e))

@app.post("/deploy/")
async def deploy_model(input_data: DeployInput):
    logging.info(f"Received request to deploy '{input_data.deployment_name}'")
    mii_configs = {
        "tensor_parallel": input_data.tensor_parallel,
        "dtype": input_data.dtype,
        "enable_restful_api": True,
        "restful_api_port": input_data.restful_api_port,
        "hostfile": "./hostfile",
    }
    logging.info(f"Deploying '{input_data.deployment_name}'")
    mii.deploy(task="text-generation",
               model=input_data.model,
               deployment_name=input_data.deployment_name,
               mii_config=mii_configs
               )
    try:
        logging.info(f"Creating generator for deployment '{input_data.deployment_name}'")
        generator = mii.mii_query_handle(input_data.deployment_name)
        # deployed_models[input_data.deployment_name] = generator
        logging.info(f"Deployment '{input_data.deployment_name}' successful")
        response = {"message": f"Deployment '{input_data.deployment_name}' successful"}
        return response

    except Exception as e:
        logging.error(f"Deployment '{input_data.deployment_name}' failed")
        raise HTTPException(status_code=500, detail=str(e))
```

My hostfile contents:
`localhost slots=8`

I am using an `ec2 instance of g5.48xlarge which has 8 gpu's`. I would love to provide more details if needed. Thanks

**Comment by mrwyattii** [comment]
@msinha251 Thanks for reporting this bug. We have a fix for this in #192. Could you try with this PR? If it works for you, we will get it merged and make a new release of MII. Thanks!

**Comment by mrwyattii** [comment]
Fix has been merged, closing
